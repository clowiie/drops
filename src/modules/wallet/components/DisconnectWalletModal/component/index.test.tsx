import React from 'react'

import { fireEvent, render } from '@testing-library/react'

import mockComponent from 'test/utils/mockComponent'

import * as DisconnectWalletModalType from '.'

describe('DisconnectWalletModal', () => {
  const ModalSpy = mockComponent('common/components/Modal')

  const useDisconnectWalletModalSpy = jest.fn()
  const disconnectWalletSpy = jest.fn()
  jest.doMock(
    'modules/wallet/hooks/useDisconnectWalletModal',
    () => useDisconnectWalletModalSpy,
  )

  const { default: DisconnectWalletModal } =
    require('.') as typeof DisconnectWalletModalType

  beforeEach(() => {
    useDisconnectWalletModalSpy.mockReturnValue({
      disconnectWallet: disconnectWalletSpy,
    })
  })

  afterEach(() => {
    jest.clearAllMocks()
  })

  describe('snapshots', () => {
    it('should match snapshots', () => {
      const { container, getByText } = render(<DisconnectWalletModal isOpen />)

      expect(container.firstChild).toMatchSnapshot()
      expect(getByText('Disconnect')).toBeDefined()
      expect(getByText('Logout')).toBeDefined()
      expect(ModalSpy).toBeCalledTimes(1)
    })

    it('should match snapshots with isOpen is false', () => {
      const { container } = render(<DisconnectWalletModal isOpen={false} />)

      expect(container.firstChild).toMatchSnapshot()
      expect(ModalSpy).toBeCalledTimes(1)
    })
  })

  describe('render component', () => {
    it('should call disconnectWallet when press logout button', () => {
      const { getByText } = render(<DisconnectWalletModal isOpen />)

      const button = getByText('Logout')

      fireEvent.click(button)

      expect(disconnectWalletSpy).toBeCalledTimes(1)
    })
  })
})
