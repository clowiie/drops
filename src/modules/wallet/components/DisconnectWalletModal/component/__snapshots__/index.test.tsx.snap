// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisconnectWalletModal snapshots should match snapshots 1`] = `
<div>
  <div
    class="p-4 flex justify-between items-center"
  >
    <span
      class="text-base"
    >
      Disconnect
    </span>
    <button
      class="bg-gray-700 hover:bg-gray-500 text-white p-2 rounded-lg text-sm"
    >
      Logout
    </button>
  </div>
</div>
`;

exports[`DisconnectWalletModal snapshots should match snapshots with isOpen is false 1`] = `
<div>
  <div
    class="p-4 flex justify-between items-center"
  >
    <span
      class="text-base"
    >
      Disconnect
    </span>
    <button
      class="bg-gray-700 hover:bg-gray-500 text-white p-2 rounded-lg text-sm"
    >
      Logout
    </button>
  </div>
</div>
`;
